azurerm/resource_arm_azure_firewall_network_rule_collection.go

```diff
@@ -0,0 +1,355 @@
+		return fmt.Errorf("Error making Read request on Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+	_, exists := findArmAzureFirewallNetworkRuleCollectionByName(&firewall, name)
+	if !exists {
+		return nil
+	}
+	updatedCollections := removeArmAzureFirewallNetworkRuleCollectionByName(&firewall, name)
+	firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections = updatedCollections
+
+	ipConfigs := fixArmAzureFirewallIPConfiguration(&firewall)
+	firewall.AzureFirewallPropertiesFormat.IPConfigurations = &ipConfigs
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, firewallName, firewall)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for creation/update of Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func expandArmAzureFirewallNetworkRuleCollection(d *schema.ResourceData) network.AzureFirewallNetworkRuleCollection {
+	name := d.Get("name").(string)
+	action := network.AzureFirewallRCActionType(d.Get("action").(string))
+	rules := expandArmAzureFirewallNetworkRules(d)
+	properties := network.AzureFirewallNetworkRuleCollectionPropertiesFormat{
+		Action: &network.AzureFirewallRCAction{
+			Type: action,
+		},
+		Priority: utils.Int32(int32(d.Get("priority").(int))),
+		Rules:    &rules,
+	}
+	col := network.AzureFirewallNetworkRuleCollection{
+		Name: &name,
+		AzureFirewallNetworkRuleCollectionPropertiesFormat: &properties,
+	}
+	return col
+}
+
+func findArmAzureFirewallNetworkRuleCollectionByName(firewall *network.AzureFirewall, name string) (*network.AzureFirewallNetworkRuleCollection, bool) {
+	for _, collection := range *firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections {
+		if collection.Name != nil && *collection.Name == name {
+			return &collection, true
+		}
+	}
+	return nil, false
+}
+
+func removeArmAzureFirewallNetworkRuleCollectionByName(firewall *network.AzureFirewall, name string) *[]network.AzureFirewallNetworkRuleCollection {
+	var index int
+	collections := *firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections
+	for i, collection := range collections {
+		if collection.Name != nil && *collection.Name == name {
+			index = i
+			continue
+		}
+	}
+	collections = append(collections[:index], collections[index+1:]...)
+	return &collections
+}
+
+func expandArmAzureFirewallNetworkRules(d *schema.ResourceData) []network.AzureFirewallNetworkRule {
+	nwRules := d.Get("rule").(*schema.Set).List()
+	rules := make([]network.AzureFirewallNetworkRule, 0)
+
+	for _, nwRule := range nwRules {
+		rule := nwRule.(map[string]interface{})
+
+		name := rule["name"].(string)
+		ruleToAdd := network.AzureFirewallNetworkRule{
+			Name: &name,
+		}
+		if description := rule["description"].(string); description != "" {
```

Comment:
I think we're safe to assign this regardless (since if the key's nil in TF, we should reset the description?)